Src/FluentAssertions/Formatting/MultidimensionalArrayFormatter.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using FluentAssertions.Common;
+
+namespace FluentAssertions.Formatting
+{
+    public class MultidimensionalArrayFormatter : IValueFormatter
+    {
+        /// <summary>
+        /// Indicates whether the current <see cref="IValueFormatter"/> can handle the specified <paramref name="value"/>.
+        /// </summary>
+        /// <param name="value">The value for which to create a <see cref="System.String"/>.</param>
+        /// <returns>
+        /// <c>true</c> if the current <see cref="IValueFormatter"/> can handle the specified value; otherwise, <c>false</c>.
+        /// </returns>
+        public bool CanHandle(object value)
+        {
+            return value is Array arr && arr.Rank >= 2;
+        }
+
+        /// <inheritdoc />
+        public string Format(object value, FormattingContext context, FormatChild formatChild)
+        {
+            var arr = (Array)value;
+
+            if (arr.Length <= 0)
+            {
+                return "{empty}";
+            }
+
+            var sb = new StringBuilder();
+
+            var indecies = Enumerable.Range(0, arr.Rank).Select(dimention => arr.GetLowerBound(dimention)).ToArray();
+
+            bool IsFirstIteration(int index, int dimention)
```

Comment:
Please don't use local functions. This method is already extremely long and quite difficult to follow.